scripts/tools/zap_regen_all.py

```diff
@@ -113,6 +144,39 @@ def generate(self) -> TargetRunStats:
         )
 
 
+class GoldenTestImageTarget():
+    def __init__(self):
+        # NOTE: output-path is inside the tree. This is because clang-format
+        #       will search for a .clang-format file in the directory tree
+        #       so attempts to format outside the tree will generate diffs.
+        os.makedirs('./out', exist_ok=True)
+        self.tempdir = tempfile.mkdtemp(prefix='test_golden', dir='./out')
+        self.command = ["./scripts/tools/zap/test_generate.py", "--output", self.tempdir, "--regenerate"]
```

Comment:
This is worth a comment that the important part here is that `--regenerate` will cause the existing test outputs to be replaced with a copy of whatever gets generated into the temp dir, right?

Reply:
Added comment.